resource-aggregate: skip duplicate deviceMetadataUpdated events

Danielius1922 opened the pull request:
Skip the event being handled if its equivalent to the stored
snapshot.

Danielius1922 committed:
resource-aggregate: skip duplicate deviceMetadataUpdated events

Skip the event being handled if its equivalent to the stored
snapshot.

jkralik commented on resource-aggregate/events/deviceMetadataUpdated.go:
```diff
@@ -34,3 +34,12 @@ func (e *DeviceMetadataUpdated) GroupID() string {
+		e.Status.ValidUntil == upd.Status.ValidUntil &&
+		e.ShadowSynchronization == upd.ShadowSynchronization &&
+		e.AuditContext.UserId == upd.AuditContext.UserId &&
+		e.AuditContext.CorrelationId == upd.AuditContext.CorrelationId
```
Pls use getters functions instead of direct access.

Danielius1922 committed:
resource-aggregate: fix segfault

Use Getters for accessing ResourceChanged event properties
to avoid segmentation fault on nil data.
